core/renderers/common/path_object.js

```diff
@@ -74,6 +67,27 @@ Blockly.blockRendering.PathObject = function(root) {
   this.svgPathDark = Blockly.utils.dom.createSvgElement('path',
       {'class': 'blocklyPathDark', 'transform': 'translate(1,1)'},
       this.svgRoot);
+
+  /**
+   * Primary colour of the block in '#RRGGBB' format.
+   * @type {string}
+   * @package
+   */
+  this.primaryColour = '#000000';
+
+  /**
+   * Secondary colour of the block in '#RRGGBB' format.
+   * @type {string}
+   * @package
+   */
+  this.secondaryColour = '#000000';
+
+  /**
+   * Tertiary colour of the block in '#RRGGBB' format.
+   * @type {string}
+   * @package
+   */
+  this.tertiaryColour = '#000000';
```

Comment:
Yes, because the path object is passed the style to set these, so we could add more properties.

That said, it might be good to store the style as well.  I'll noodle on that.